cmd/contour/serve.go

```diff
@@ -184,10 +185,10 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		AccessLogFields:       ctx.AccessLogFields,
 		MinimumTLSVersion:     annotation.MinTLSVersion(ctx.TLSConfig.MinimumProtocolVersion),
 		RequestTimeout:        ctx.RequestTimeout,
-		ConnectionIdleTimeout: ctx.ConnectionIdleTimeout,
-		StreamIdleTimeout:     ctx.StreamIdleTimeout,
-		MaxConnectionDuration: ctx.MaxConnectionDuration,
-		DrainTimeout:          ctx.DrainTimeout,
+		ConnectionIdleTimeout: timeout.Parse(ctx.ConnectionIdleTimeout),
```

Comment:
What happens if someone passes an invalid config option, it's disabled?

Reply:
Currently yes, it would be disabled, which is consistent with how timeouts specified in HTTPProxy are treated today. But there's discussion in #2697 about bubbling these up as errors instead, which I think makes more sense.

Reply:
Fantastic! :tada: